utils/docker/Dockerfile.centos.7

```diff
@@ -12,13 +12,69 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# Switch to using DNF instead of yum.
+#
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories.
+# The Docker image starts out with the DISTRO provided GPG keys
+# on the disk, but not installed.  These must be isntalled before
+# adding any repo that needs GPG keys but does not provide meta
+# data for installing those keys.  Then the epel-release package
+# version from the extras repository must be installed and the
+# GPG key it provides must be installed.
+ARG REPO_URL=""
+ARG REPO_DISTRO=""
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/* ;                         \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/" \
+          -i /etc/yum.conf;                                     \
+      yum -y install --disablerepo=base,extras,updates          \
+             dnf dnf-plugins-core epel-release-7-11;            \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7;         \
+      dnf config-manager --assumeyes --quiet                    \
+          --disable base epel extras updates;                   \
+    else                                                        \
+      yum -y install dnf dnf-plugins-core epel-release-7-11;    \
+    fi;                                                         \
+    dnf -y upgrade epel-release;                                \
+    yum clean all;                                              \
+    dnf clean all
+
+# If a local DAOS repository is supplied, then we should use it
+# This is mainly for quickbuilds
```

Comment:
Now that CI re-uses the images across builds so only does a OS install or dependencies build if they've changed, then I'm not sure of the benefit of quickbuild in the docker builds.  By specifying "Parallel-build: true" the docker builds can finish in about 60-90 seconds so maintaining the additional complexity of quickbuild in the docker images doesn't seem to add much.